app/models/user.rb

```diff
@@ -456,6 +456,16 @@ def close_account!
     AccountDeletion.create(:person => self.person)
   end
 
+  def lock_account!
+    self.locked_at = Time.now
+    self.save
+  end
```

Comment:
@jhass Devise is a separate module. Don't we need to write the code in core repository? Kindly clarify it.